src/Symfony/Component/Intl/Data/Generator/LocaleDataGenerator.php

```diff
@@ -138,69 +138,29 @@ public function generateData(GeneratorConfig $config)
 
     private function generateLocaleName($locale, $displayLocale)
     {
-        $name = null;
-
-        $lang = \Locale::getPrimaryLanguage($locale);
-        $script = \Locale::getScript($locale);
-        $region = \Locale::getRegion($locale);
-        $variants = \Locale::getAllVariants($locale);
-
         // Currently the only available variant is POSIX, which we don't want
         // to include in the list
-        if (\count($variants) > 0) {
-            return;
-        }
-
-        // Some languages are translated together with their region,
-        // i.e. "en_GB" is translated as "British English"
-        // we don't include these languages though because they mess up
-        // the name sorting
-        // $name = $this->langBundle->getLanguageName($displayLocale, $lang, $region);
-
-        // Some languages are not translated
-        // Example: "az" (Azerbaijani) has no translation in "af" (Afrikaans)
-        if (null === ($name = $this->languageDataProvider->getName($lang, $displayLocale))) {
```

Comment:
null API is not applicable, it throws instead. This is try/catched on the calling side already

Reply:
besides, this case is translated in fact :D